homeassistant/components/google_assistant/smart_home.py

```diff
@@ -113,6 +113,24 @@ def entity_to_device(entity: Entity):
 
 def query_device(entity: Entity) -> dict:
     """Take an entity and return a properly formatted device object."""
+    if entity.domain == climate.DOMAIN:
+        mode = entity.attributes.get(climate.ATTR_OPERATION_MODE)
+        response = {
+            'thermostatMode':
+            mode if mode in CLIMATE_SUPPORTED_MODES else 'on',
+            'thermostatTemperatureSetpoint':
+            entity.attributes.get(climate.ATTR_TEMPERATURE),
+            'thermostatTemperatureAmbient':
+            entity.attributes.get(climate.ATTR_CURRENT_TEMPERATURE),
+            'thermostatTemperatureSetpointHigh':
+            entity.attributes.get(climate.ATTR_TARGET_TEMP_HIGH),
+            'thermostatTemperatureSetpointLow':
+            entity.attributes.get(climate.ATTR_TARGET_TEMP_LOW),
+            'thermostatHumidityAmbient':
+            entity.attributes.get(climate.ATTR_CURRENT_HUMIDITY),
+        }
+        return {k: v for k, v in response.items() if v is not None}
+
     final_state = entity.state != STATE_OFF
```

Comment:
While you are here, I suggest wrapping the lines about state and brightness into `if entity.domain = FOO` as well. There is no point trying to return state for domains to which those states do not apply.

Reply:
@infernix It's a good suggestion and I agree that the code for different domains should be clearly delineated throughout the google_assistant component, that cleanup is unrelated to this PR and I don't want to hold it back while I go sort things out (there's no documentation beyond the code itself showing which domains are supported and which states they expose).